Jk admin new layout

kulincz opened the pull request:
| Q             | A
| ------------- | ---
|Description, reason for the PR| Top menu is moved to the right panel and layout is fluid for whole screen
|New feature| No <!-- Do not forget to update CHANGELOG.md and possibly docs/ -->
|BC breaks| Yes/No <!-- Do not forget to update UPGRADE.md -->
|Fixes issues| ...
|Standards and tests pass| Yes/No
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes/No

LukasHeinz commented:
Comments form business POV:
1) during rebranding we have to handle colors of domain subheadings (see image below)
<img width="990" alt="snimek obrazovky 2018-06-20 v 23 54 28" src="https://user-images.githubusercontent.com/33071107/41686724-6b741c88-74e5-11e8-94ad-401f92f64873.png">
2) I do not like color of buttons - we need to discuss it
3) in menu when I click to some section, whole menu will collaps and then the selected section expand again. This is really irritaiting
4) when I click on setting in menu, I see no change because the submenu expanded below - it should be resolved somehow
5) active item in menu is hardly distinguished - I suggest to highlight it a bit more
6) when Orders and Customers menu items have only one subitem - would it make sense to use it as only item like dashboard (so no submenu)?

kulincz commented:
3. this issue is fixed.

kulincz commented:
1. Fixed.
2. It's brand color of SSFW. We can do it different. I was traying to use light blue brand color, but it was all cold (only blue colors) and does't work well as CTA color. My suggestion is to stay with orange and maybe think about inverse button with dark text and slightly orange background. 
3. Fixed.
4. I will try to add some space under the last item in the menu – so user than will see that something appear after his click.
5. Ok, I will fix this.
6. I agree. This will be much better. This issue is for PRG I think.

PetrHeinz commented:
We have fixed the redesign from the programmer's point of view and simplified things with @kulincz. Requesting code review, so we can merge it as soon as possible (testing and BV will be needed, of course).

PetrHeinz commented:
@kulincz: I've found out that the detail of a product still has the old logo, could you please fix it?

![image](https://user-images.githubusercontent.com/10008612/41980371-9edc0c16-7a26-11e8-9000-814529ae111e.png)

PetrHeinz commented:
@kulincz: there are two more things I've noticed:

1) The inlined SVG seems to me like a bad idea, it's not readable at all. Can it be included as a resource similarly to an image?
https://github.com/shopsys/shopsys/blob/35fc1c1356d86343516879830697cdb929b84199/packages/framework/src/Resources/views/Admin/Layout/layout.html.twig#L20-L55

2) Two defined colors remained unchanged in the LESS: `@color-red` and `@color-yellow`. Is it OK to use those colors after rebranding?

MattCzerner commented on packages/framework/src/Resources/config/admin_menu.yml:
```diff
@@ -6,26 +6,20 @@
-        - label: 'Editing order'
-          route: admin_order_edit
-          visible: false
+    - label: 'Editing order'
```
I wonder if we should consider these items as MenuItem, it is not displayed in menu and href is in template. Is it because of breadCrumb or why?

PetrHeinz replied:
Exactly because of Breadcrumb. I agree it's weird but solving it is out of scope of this redesign.

MattCzerner commented on CHANGELOG.md:
```diff
@@ -42,6 +42,12 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
+    - colors changed to match new Shopsys CI
+    - main menu moved to the left panel along with settings menu
+    - menu items do not have type anymore, which lead to simplification of the code
+    - menu was restructured
```
third level is disabled?

PetrHeinz replied:
Commit 94a1384601656617b89f1208c9aceafc820652f9 dropped. I've slightly reordered some items after a consultation with @LukasHeinz in d85ff8779.

kulincz commented:
@LukasHeinz 
4. Fixed here: https://github.com/shopsys/shopsys/commit/d0f12811d15a529ea5e6efc8db895463335516e8
5. Fixed here: https://github.com/shopsys/shopsys/commit/62eaaaa224923a9e2cac8f0013695b1b71aee775
6. Fixed

@PetrHeinz 
Logo on edit product page Fixed here: https://github.com/shopsys/shopsys/commit/a7ff239ce231ad3cfd4a590739d176bea3a3f939
1. Fixed here: https://github.com/shopsys/shopsys/commit/79ec2ca9c2541d34f58d2fe5b7b7748a367bd461
2. I will check it. Thank for reminder.

PetrHeinz commented:
@kulincz thanks for the fixes. Please, add those commits to the branch you are trying to merge in this PR. Work elsewhere cannot be reviewed and merged.

Your commit a7ff239ce231ad3cfd4a590739d176bea3a3f939 still contains TABs instead of spaces. Please, check your commits on Github when submitting them for others to work with.

PetrHeinz commented on packages/framework/src/Resources/views/Admin/Layout/layout.html.twig:
```diff
@@ -16,6 +16,45 @@
             <header class="web__header">
                 <div class="web__header__in">
+                    <div class="web__header__logo">
+                        <svg class="web__header__logo__svg" viewBox="0 0 701 208" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
```
Done in 79ec2ca. Waiting for @kulincz to put it into this PR.

PetrHeinz commented on packages/framework/src/Controller/Admin/MenuController.php:
```diff
@@ -26,7 +26,7 @@ public function __construct(MenuFactory $menuFactory, DomainFacade $domainFacade
     public function menuAction($route, array $parameters = null)
     {
-        $menu = $this->menuFactory->createMenuWithVisibleItems();
+        $menu = $this->menuFactory->createMenuWithAllowedItems();
```
Exactly, it's just renaming.

PetrHeinz commented on packages/framework/src/Model/AdminNavigation/MenuFactory.php:
```diff
@@ -79,19 +79,19 @@ private function filterVisibleRecursive(array $menuItems)
      * @return bool
      */
-    private function isMenuItemVisible(MenuItem $menuItem)
+    private function isMenuItemAllowed(MenuItem $menuItem)
```
It's not just authorization - `multidomain_only` is allowed only when having multimple domains. The "switch" is basically the configuration in `admin_menu.yml`.

I would personally leave it as is, but I'll appreciate if you come up with a better name.

PetrHeinz commented on packages/framework/src/Resources/config/admin_menu.yml:
```diff
@@ -180,98 +180,88 @@
-          route: admin_country_list
+        - label: 'Editing brand'
+          route: admin_brand_edit
+          visible: false
```
The items are there because of Breadcrumb. There was always some hard-coded maximum level. Previously only second level was rendered in the left menu, now there have been two levels. Changed to three levels as I've dropped the commit based on your and @LukasHeinz's feedback.

This is something we will probably address in the future.

PetrHeinz commented on packages/framework/src/Resources/config/admin_menu.yml:
```diff
@@ -180,98 +180,88 @@
-          route: admin_superadmin_urls
-    - label: 'External scripts'
+
+    - label: 'Error messages'
```
*URL addresses* are missing the postfix as well, see http://127.0.0.1:8000/admin/superadmin/urls/

The order wasn't changed, let's address it in the future along with the other menu issues.

PetrHeinz commented:
Rebased on `master` with changes requested by @MattCzerner and @LukasHeinz.

Waiting for the @kulincz's fixes, then on the way to final review.

PetrHeinz commented:
We've managed to fix it up with @kulincz :)

@MattCzerner, please, review it after the fixes. If there is still some issue or if you have a question, please, call me - let's resolve this as soon as possible. If it'S OK, send it to tests and make sure it can be submitted to BV till this evening.

Thanks a lot :+1:

MattCzerner commented on packages/framework/src/Resources/styles/admin/layout/header/top-info.less:
```diff
@@ -1,32 +1,31 @@
 .top-info {
```
In this whole file, tabs are used instead of whitespaces. Why?

PetrHeinz replied:
It took a while for kulincz to configure his IDE, I thought I've fixed up everything... Fixed.

PetrHeinz commented:
The inability to collapse menu was intentional, although I'm not sure why to be honest. We can tweak it later :)

PetrHeinz commented:
@MattCzerner has it been tested?

LukasHeinz commented:
Testing report:
1) MacOS & Chrome - Dashboard has epileptic seizure. For more please see https://drive.google.com/file/d/1UV4y3-CSNqfgaAd2UQGChlY3dNussDKN/view?usp=sharing
2) MacOS (Ch and FF) - there is horizontal scrollbar in whole admin for no reason
<img width="1280" alt="snimek obrazovky 2018-06-29 v 23 20 32" src="https://user-images.githubusercontent.com/33071107/42115443-2a741566-7bf3-11e8-8b2a-37a3457e8d94.png">
3) Text on CTA buttons is problematic to read - I suggest color darker or place some black border around text to make it more contrast.

kulincz commented:
1. 2. Should be fixed.
3) I decide to change text color to black – there is way more better readability.

jakubhavlak commented:
Testing report no.2 (tested on Win & Chrome):

1. When I click to add new product on main page the "Finish the change" button is not fully visible in pop up window. 
Location: Admin - Marketing - Main page products
![image](https://user-images.githubusercontent.com/31988844/42161712-4a46bbf4-7dfc-11e8-95f5-c3ad0aab4898.png)


2. Default background is little bit visible on the left side of the colour square when editing the flag colour.
Location: Admin - Settings - Lists and nomenclatures - Flags overview 
![image](https://user-images.githubusercontent.com/31988844/42161556-b35a3b26-7dfb-11e8-8729-cb8680953e36.png)
